include/eve/module/core/regular/impl/negmaxabs.hpp

```diff
@@ -9,63 +9,25 @@
 
 #include <eve/concept/compatible.hpp>
 #include <eve/concept/value.hpp>
-#include <eve/detail/apply_over.hpp>
-#include <eve/detail/implementation.hpp>
-#include <eve/detail/skeleton_calls.hpp>
-#include <eve/module/core/regular/abs.hpp>
-#include <eve/module/core/regular/all.hpp>
-#include <eve/module/core/regular/is_nan.hpp>
-#include <eve/module/core/regular/is_not_greater_equal.hpp>
 #include <eve/module/core/regular/maxabs.hpp>
-#include <eve/module/core/regular/minus.hpp>
-#include <eve/traits/common_compatible.hpp>
 
 #include <type_traits>
 
 namespace eve::detail
 {
-template<real_value T, real_value U>
-EVE_FORCEINLINE auto
-negmaxabs_(EVE_SUPPORTS(cpu_), T const& a, U const& b) noexcept requires compatible_values<T, U>
-{
-  return arithmetic_call(negmaxabs, a, b);
-}
 
-template<real_value T>
+template<typename  ...Ts>
 EVE_FORCEINLINE auto
-negmaxabs_(EVE_SUPPORTS(cpu_), T const& a, T const& b) noexcept requires has_native_abi_v<T>
+negmaxabs_(EVE_SUPPORTS(cpu_), Ts... args)
 {
-  return minus(maxabs(a, b));
+  return minus(maxabs(args...));
 }
 
-//================================================================================================
-// Masked case
-//================================================================================================
-template<decorator D, conditional_expr C, real_value U, real_value V>
+template<conditional_expr C, typename  ...Ts>
 EVE_FORCEINLINE auto
-negmaxabs_(EVE_SUPPORTS(cpu_), C const& cond, D const&, U const& t, V const& f) noexcept requires
-    compatible_values<U, V>
+negmaxabs_(EVE_SUPPORTS(cpu_), C const & c, Ts... args)
 {
-  return mask_op(cond, D()(eve::negmaxabs), t, f);
+  return minus[c](maxabs[c](args...));
```

Comment:
doesn't it perform two `if_else` ?

Reply:
yes when the intrinsic is not used. Is it problematic ?